### DISCUSSION TITLE
Should the order of literals be hex/oct/dec when parsing an integer?

### DISCUSSION THREAD

**Discussion by 0x8000-0000** [taocpp/PEGTL#279] [Q&A] [answered]
Regarding:

https://github.com/taocpp/PEGTL/blob/45b4d68dae5c231666b696a0955eb2fa871ef2e2/src/example/pegtl/proto3.cpp#L46

I tried using this fragment in one of my grammars, and hex numbers failed to parse, the leading 0 was recognized as a decimal then the parsing failed. Switching the order:

     struct int_lit : sor< hex_lit, oct_lit, dec_lit > {};

seems to have solved the problem

**Comment by d-frey** [comment, marked as answer]
Yes. I fixed it, but after looking at the proto3 grammar I think there are more things wrong with our grammar. *sigh* I'll take a look when I have some more time.

**Reply by 0x8000-0000** [reply]
Thank you!
